### PR TITLE
SIMSBIOHUB 335: Tab Through Observation Edit

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## Links to Jira Tickets

- [SIMSBIOHUB-335](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-335)

## Description of Changes

- Modified each custom renderEditCell component of the ObservationTable to manually set hasFocus. This allows the user to Tab forward to the next cell or Shift+Tab back to the previous. It seems that if you don't override the renderEditCell, it does this by default, but it is necessary for any custom component. [Docs here.](https://mui.com/x/react-data-grid/editing/#create-your-own-edit-component)
- Wrapped some of the simpler TextField / DatePicker renders in their own functional components so that the same hasFocus pattern could be used.

### CODE REVIEW

File: app/src/features/surveys/observations/ObservationsTable.tsx [hunk 1]
```diff
@@ -265,25 +264,45 @@ const ObservationsTable = (props: ISpeciesObservationTableProps) => {
...
+              }
             }}
-            value={params.value ?? ''}
-            variant="outlined"
-            type="text"
-            inputProps={{ inputMode: 'numeric' }}
           />
         );
+
+        // return (
```
Comment (al-rosenthal) [inline app/src/features/surveys/observations/ObservationsTable.tsx]: nit: we could probably remove these commented code blocks

File: app/src/components/data-grid/autocomplete/AsyncAutocompleteDataGridEditCell.tsx [hunk 2]
```diff
@@ -52,6 +53,14 @@ const AsyncAutocompleteDataGridEditCell = <DataGridType extends GridValidRowMode
 
   const apiRef = useGridApiContext();
 
+  const ref = useRef<HTMLInputElement>();
+
+  useEnhancedEffect(() => {
+    if (dataGridProps.hasFocus) {
+      ref.current?.focus();
+      console.log('Focus fired!');
```
Comment (al-rosenthal) [inline app/src/components/data-grid/autocomplete/AsyncAutocompleteDataGridEditCell.tsx]: nit: remove pls

File: app/src/components/data-grid/autocomplete/AutocompleteDataGridEditCell.tsx [hunk 3]
```diff
@@ -45,6 +47,15 @@ const AutocompleteDataGridEditCell = <DataGridType extends GridValidRowModel, Va
 
   const apiRef = useGridApiContext();
 
+  const ref = useRef<HTMLInputElement>();
+
+  useEnhancedEffect(() => {
+    if (dataGridProps.hasFocus) {
+      ref.current?.focus();
+      console.log('Focus fired!');
```
Comment (al-rosenthal) [inline app/src/components/data-grid/autocomplete/AutocompleteDataGridEditCell.tsx]: nit: remove pls

**Review by al-rosenthal — COMMENTED** [review]
ran this locally and works great

**Review by curtisupshall — APPROVED** [review]
Code looks good to me

**Review by JeremyQuartech — APPROVED** [review]
Tested locally and it works perfectly 👍

**Review by al-rosenthal — APPROVED** [review]
👨‍🍳 💋

**Pull request merged**